app/javascript/multi-stamp-tubes/components/MultiStampTubes.spec.js

```diff
@@ -268,8 +268,8 @@ describe('MultiStampTubes', () => {
     expect(wrapper.vm.targetWells).toEqual({
       A1: { pool_index: 1 },
       B1: { pool_index: 2 },
-      C1: { pool_index: 3 },
-      D1: { pool_index: 4 },
+      C1: { pool_index: 2 },
+      D1: { pool_index: 3 },
     })
   })
```

Comment:
Wondered if there were tests for the individual Vue components.

Also wondered what happens when they remove the last instance of a barcode from the scan boxes, does it free up that colour index to be used for the next new unique barcode?

Reply:
It probably shouldn't right, if that one they delete was in the middle of the colour list you don't want complications with latter ones being re-coloured.

Reply:
_Deleting the barcode will cause all other codes to be shifted up by one._

As in it shifts them up into new rack locations? I think that would be bad if that's what you mean, as they are probably physically placing the tubes in a rack as they scan them.

Reply:
If you mean just shifted in colour index then that's acceptable I think. The colours are just a visual aid to see where the replicates are.

Reply:
I noticed that functionality too, the colours changing when you remove a barcode. I thought it was ok.